server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java

```diff
@@ -198,6 +198,58 @@ public void testParsingStrictNanoDates() {
         formatter.format(formatter.parse("2018-05-15T17:14:56.123456789+01:00"));
     }
 
+    public void testIso8601Parsing() {
+        DateFormatter formatter = DateFormatters.forPattern("iso8601");
+
+        // timezone not allowed with just date
+        formatter.format(formatter.parse("2018-05-15"));
+
+        formatter.format(formatter.parse("2018-05-15T17"));
```

Comment:
Are we sure we need this? I am not sure if iso8601 is really specifying this all, but java.time'` DateTimeFormatter.ISO_DATE_TIME is a bit more strict. it would fail for this and many others:
```
DateTimeFormatter.ISO_DATE_TIME.parse("2018-05-15T17")
DateTimeFormatter.ISO_DATE_TIME.parse("2018-05-15T17+01:00")
etc..
```